e2e_testing/xfail_sets.py

```diff
@@ -688,7 +688,11 @@
     "FullLikeModuleInt2DStatic_basic",
     "FullModuleInt3D_basic",
     "FullModuleFloat2D_basic",
-    "RepeatModule_basic"
+    "RepeatModule_basic",
+    "ConstantPad2dStaticModule_basic",
+    "ConstantPadNdModule_basic",
+    "ConstantPadNdPartialStaticModule_basic",
+    "ConstantPadNdStaticModule_basic"
```

Comment:
Can you confirm that `PadModule` and `PadWithNoneValModule` still fail? If they now pass, you should also add them (or your PR will fail testing).

Reply:
both PadModule_basic & PadWithNoneValModule_basic tests pass with this PR. I've added them to TOSA_PASS_SET.
Regarding the test/Conversion/TorchToTosa/constant_pad_nd.mlir, the only thing that makes it worth keeping is the padding value definition below. 
%float-Inf = torch.constant.float 0xFFF0000000000000

It's not a normal torch.constant.float. It's a torch float that comes from an INT representation. That's why I need to add 

Operation *padOp = padValue.getDefiningOp();
padValue = padOp->getOperand(0);

to extract the float value before calling torchScalarTosaTensor. So, maybe I just add that test to the basir.mlir?